lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -7134,12 +7134,12 @@ namespace Js
 
     SourceTextModuleRecord* JavascriptLibrary::GetModuleRecord(uint moduleId)
     {
-        Assert((moduleRecordList->Count() >= 0) && (moduleId < (uint)moduleRecordList->Count()));
-        if (moduleId >= (uint)moduleRecordList->Count())
+        Assert((moduleRecordList->Count() > 0) && (moduleId <= (uint)moduleRecordList->Count()));
+        if (moduleId > (uint)moduleRecordList->Count())
         {
             Js::Throw::FatalInternalError();
         }
-        return moduleRecordList->Item(moduleId);
+        return moduleRecordList->Item(moduleId - 1); // 0 == kmodGlobal
```

Comment:
Why use kmodGlobal?  That is the value to distingish between the different global scopes (the real global scope of form global scope).  It has nothing to do with Javascript modules)
Please define a different constant to mean: this is not a module?  May be use the value -1 instead os you don't have to do this weird shifting?